shim/src/handle_vm_kvm_create_irqchip.c

```diff
@@ -48,6 +49,13 @@ handle_vm_kvm_create_irqchip(struct shim_vm_t *const pmut_vm) NOEXCEPT
         bferror("The shim is not running in a VM. Did you forget to start MicroV?");
         return SHIM_FAILURE;
     }
+
+    platform_mutex_lock(&pmut_vm->mutex);
+    if (pmut_vm->is_irqchip_created) {
+        bferror("The IRQCHIP is not created, Did you forget to create it?");
+        return SHIM_FAILURE;
```

Comment:
We should have a `mut_ret` variable set to `SHIM_FAILURE` and a `goto mutex_unlock` statement here.

Reply:
So here the idea is unlock mutex when irqchip is created, how if it is as below

![image](https://user-images.githubusercontent.com/53925815/155278458-54a31a8f-398e-4fdc-acdd-59264ebf493a.png)